aspnet/security/anti-request-forgery.md

```diff
@@ -0,0 +1,85 @@
+
+When cookie  is used, The authentication cookie is just a container for the forms authentication ticket. The ticket is passed as the value of the forms authentication cookie with each request and is used by forms authentication, on the server, to identify an authenticated user.
+
+When a user is logged in to a system, a user session is created on the server side and it is stored in the database or in some kind of in-memory storage systems. The system generates a session key that points to the actual session in the database and it is attached to a client side cookie. After that every time when a user asks for a page which needs authorization, the session key in the cookie is finds the appropriate user session in the database. Then the system checks if the user from whom the user session is stored has the privilege to access that page. If it does, then eventually the application takes the user to that page. As you can see we are using cookies to make the application stateful. USERTOKENS Token based authentication on the other hand doesn’t store any kind of session in the database. Instead when a user is logged in he is issued with a token (not antiforgery token). This token is self-contained. It contains all the data that is required to validate the token as well
+as user information through claims. When a user wants to access some authorized pages the token is send to the server with an additional authorization header in form of Bearer {token} (Recall from previous web api applications with single user authentication enabled). This makes the application stateless since in every subsequent request the token is passed in the request for server side validation. One thing to remember is this token is not encrypted rather it is encoded. On the server side the token can be decoded to have the raw information about the token. To send the token in subsequent requests, you can either store it in browser’s local storage or in a cookie. You don’t have to worry about anything related to XSRF if your token is stored in the local storage. But you have to deal with XSRF if you store it in a cookie.
+
+  ### Multiple applications are hosted in one domain
+
+Even though `example1.cloudapp.net` and `example2.cloudapp.ne``t are different hosts, there is an implicit trust relationship between all hosts under the ``*.cloudapp.net` domain. This implicit trust relationship allows potentially untrusted hosts to affect each other’s cookies (the same-origin policies that govern AJAX requests do not necessarily apply to HTTP cookies). The ASP.NET Core runtime provides some mitigation in that the username is embedded into the field token, so even if a malicious subdomain is able to overwrite a session token it will be unable to generate a valid field token for the user. However, when hosted in such an environment the built-in anti-XSRF routines still cannot defend against session hijacking or login XSRF attacks.
```

Comment:
That's not exactly true. Cookies can be limited to hosts, and Chrome (for example) knows about cloudapp.net and azurewebsites.net being a hosting domain. and provides some extra protection